<a href="https://colab.research.google.com/github/shanchepngen-ux/ML-For-Engineering/blob/main/introduction_to_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Step One:** Mount Google Drive

In [ ]:
from google.colab import drive
drive.mount("/content/drive", force_remount=True)

Mounted at /content/drive


In [ ]:
import pandas as pd

file_path = '/content/drive/My Drive/Colab Notebooks/AmazonReviews - AmazonReviews.csv'

try:
    df = pd.read_csv(file_path)
    print(f"Successfully loaded '{file_path}' into DataFrame.")
    display(df.head())
except FileNotFoundError:
    print(f"Error: '{file_path}' not found. Please ensure the file is in the correct location and the name is exact.")
except Exception as e:
    print(f"An error occurred: {e}")

Successfully loaded '/content/drive/My Drive/Colab Notebooks/AmazonReviews - AmazonReviews.csv' into DataFrame.


,id,profileName,text,date,title,rating,images,helpful,variant:
0,R2IOYW8GNJDUAZ,G. W.,"So far I've only made it to ""for loops"" chapte...","Reviewed in the United States on December 26, ...","Recommend getting, but also get proof of Pytho...",5.0 out of 5 stars,NaN,19,NaN
1,R3NY3KXV8N46K7,Owen C. Marshall,"With the lockdown from the global pandemic, I ...","Reviewed in the United States on May 7, 2020","Easy to Follow, Good Intro for Self Learner",5.0 out of 5 stars,NaN,13,NaN
2,R270DXU8YVGFPX,Umar,"In embarking on my journey to learn Python, I ...","Reviewed in the United States on December 15, ...",Great for starting your journey on learning py...,5.0 out of 5 stars,NaN,0,NaN
3,R13Z1BSD70DMKJ,Dan,Have had this for one day and the cover is alr...,"Reviewed in the United States on July 5, 2019",Great inner content! Not that great outer qual...,4.0 out of 5 stars,NaN,341,NaN
4,R37T2H59DR3WSL,Zerono,"Wanted to add in a few tips, so beginners don'...","Reviewed in the United States on February 16, ...",Tips for beginners,5.0 out of 5 stars,NaN,10,NaN


### 1. Get Basic Information about the DataFrame

This will show us the number of entries, columns, data types, and non-null values, which is crucial for identifying potential issues like missing data or incorrect data types.

In [ ]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 338 entries, 0 to 337
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id           338 non-null    object 
 1   profileName  338 non-null    object 
 2   text         338 non-null    object 
 3   date         338 non-null    object 
 4   title        338 non-null    object 
 5   rating       338 non-null    object 
 6   images       37 non-null     object 
 7   helpful      338 non-null    int64  
 8   variant:     0 non-null      float64
dtypes: float64(1), int64(1), object(7)
memory usage: 23.9+ KB
None


### 2. Display Descriptive Statistics

For numerical columns, this provides summary statistics like count, mean, standard deviation, min, max, and quartiles. This helps in understanding the distribution of numerical data.

In [ ]:
display(df.describe(include='all'))

,id,profileName,text,date,title,rating,images,helpful,variant:
count,338,338,338,338,338,338,37,338.000000,0.0
unique,338,306,338,319,331,5,37,NaN,NaN
top,RRVFXI2H3TRZM,Amazon Customer,El contenido es exelente!\n\nEl libro me llegó...,"Reviewed in the United States on August 20, 2020",Good book,5.0 out of 5 stars,https://m.media-amazon.com/images/I/71BMeHsfGJ...,NaN,NaN
freq,1,25,1,3,3,140,1,NaN,NaN
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.106509,NaN
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,23.360049,NaN
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,NaN
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,NaN
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,NaN
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2.000000,NaN


### 3. Check for Missing Values

It's important to know if there are any missing values in your dataset and in which columns. This helps in deciding how to handle them (e.g., filling, dropping).

In [ ]:
print('Missing values per column:')
print(df.isnull().sum())

Missing values per column:
id               0
profileName      0
text             0
date             0
title            0
rating           0
images         301
helpful          0
variant:       338
dtype: int64


### 4. Drop the 'variant:' Column

Since the `variant:` column has 338 missing values (all of its entries), it provides no useful information. We can safely drop it from the DataFrame.

In [ ]:
df = df.drop(columns=['variant:'])
print("Dropped 'variant:' column. Current DataFrame columns:")
print(df.columns)

Dropped 'variant:' column. Current DataFrame columns:
Index(['id', 'profileName', 'text', 'date', 'title', 'rating', 'images',
       'helpful'],
      dtype='object')


### 5. Convert 'rating' Column to Numerical Type

The `rating` column is currently a string (e.g., '5.0 out of 5 stars'). To use it for numerical analysis, we need to extract the numerical part (e.g., '5.0') and convert it to a float.

In [ ]:
df['rating'] = df['rating'].apply(lambda x: float(x.split(' ')[0]))
print("Converted 'rating' column to float. Displaying info and first few rows:")
print(df.info())
display(df.head())

Converted 'rating' column to float. Displaying info and first few rows:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 338 entries, 0 to 337
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id           338 non-null    object 
 1   profileName  338 non-null    object 
 2   text         338 non-null    object 
 3   date         338 non-null    object 
 4   title        338 non-null    object 
 5   rating       338 non-null    float64
 6   images       37 non-null     object 
 7   helpful      338 non-null    int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 21.3+ KB
None


,id,profileName,text,date,title,rating,images,helpful
0,R2IOYW8GNJDUAZ,G. W.,"So far I've only made it to ""for loops"" chapte...","Reviewed in the United States on December 26, ...","Recommend getting, but also get proof of Pytho...",5.0,NaN,19
1,R3NY3KXV8N46K7,Owen C. Marshall,"With the lockdown from the global pandemic, I ...","Reviewed in the United States on May 7, 2020","Easy to Follow, Good Intro for Self Learner",5.0,NaN,13
2,R270DXU8YVGFPX,Umar,"In embarking on my journey to learn Python, I ...","Reviewed in the United States on December 15, ...",Great for starting your journey on learning py...,5.0,NaN,0
3,R13Z1BSD70DMKJ,Dan,Have had this for one day and the cover is alr...,"Reviewed in the United States on July 5, 2019",Great inner content! Not that great outer qual...,4.0,NaN,341
4,R37T2H59DR3WSL,Zerono,"Wanted to add in a few tips, so beginners don'...","Reviewed in the United States on February 16, ...",Tips for beginners,5.0,NaN,10


Let's list the contents of your Google Drive to help locate the `AmazonReviews.csv` file.